mysql-test/suite/galera/t/galera_undo_upgrade.cnf

```diff
@@ -0,0 +1,7 @@
+!include ../galera_2nodes.cnf
+
+[mysqld]
+wsrep_sst_method=mariabackup
+wsrep_sst_auth="root:"
+wsrep_debug=1
+innodb-undo-tablespaces=0
```

Comment:
Instead of developing a new test that currently does not work, we can slightly modify an existing test:
```diff
diff --git a/mysql-test/suite/galera/t/galera_sst_mariabackup.cnf b/mysql-test/suite/galera/t/galera_sst_mariabackup.cnf
index 857a4101406..43638ace9b3 100644
--- a/mysql-test/suite/galera/t/galera_sst_mariabackup.cnf
+++ b/mysql-test/suite/galera/t/galera_sst_mariabackup.cnf
@@ -7,9 +7,13 @@ wsrep_debug=1
 
 [mysqld.1]
 wsrep_provider_options='base_port=@mysqld.1.#galera_port;gcache.size=1;pc.ignore_sb=true'
+innodb_fast_shutdown=0
+innodb_undo_tablespaces=0
 
 [mysqld.2]
 wsrep_provider_options='base_port=@mysqld.2.#galera_port;gcache.size=1;pc.ignore_sb=true'
+innodb_fast_shutdown=0
+innodb_undo_tablespaces=3
 
 [sst]
 transferfmt=@ENV.MTR_GALERA_TFMT
```
With this, I can reproduce `Server did not transition to READY state` when the code fix is not present. When the code fix is present, there will be a couple of warnings about not being able to change `innodb_undo_tablespaces`. This is something that you would have to polish, possibly by adding one more server restart before the first step of work, or by suppressing the warning.